apps/etterlatte-vedtaksvurdering/src/main/kotlin/no/nav/etterlatte/vedtaksvurdering/VedtakBehandlingService.kt

```diff
@@ -321,30 +321,43 @@ class VedtakBehandlingService(
         behandlingId: UUID,
         brukerTokenInfo: BrukerTokenInfo,
         vedtakTilSamordning: Vedtak? = null,
-    ): VedtakOgRapid {
+    ): VedtakOgRapid? {
         logger.info("Setter vedtak til samordnet for behandling med behandlingId=$behandlingId")
         val vedtak = vedtakTilSamordning ?: hentVedtakNonNull(behandlingId)
 
-        verifiserGyldigVedtakStatus(vedtak.status, listOf(VedtakStatus.TIL_SAMORDNING))
-        val samordnetVedtakLocal =
-            repository.inTransaction { tx ->
-                repository.samordnetVedtak(behandlingId, tx = tx)
-                    .also {
-                        runBlocking {
-                            behandlingKlient.samordnet(behandlingId, brukerTokenInfo, it.id)
-                        }
+        when (vedtak.status) {
+            VedtakStatus.TIL_SAMORDNING -> {
+                val samordnetVedtakLocal =
+                    repository.inTransaction { tx ->
+                        repository.samordnetVedtak(behandlingId, tx = tx)
+                            .also {
+                                runBlocking {
+                                    behandlingKlient.samordnet(behandlingId, brukerTokenInfo, it.id)
+                                }
+                            }
                     }
-            }
 
-        return VedtakOgRapid(
-            samordnetVedtakLocal.toDto(),
-            RapidInfo(
-                vedtakhendelse = VedtakKafkaHendelseHendelseType.SAMORDNET,
-                vedtak = samordnetVedtakLocal.toDto(),
-                tekniskTid = Tidspunkt.now(),
-                behandlingId = behandlingId,
-            ),
-        )
+                return VedtakOgRapid(
+                    samordnetVedtakLocal.toDto(),
+                    RapidInfo(
+                        vedtakhendelse = VedtakKafkaHendelseHendelseType.SAMORDNET,
+                        vedtak = samordnetVedtakLocal.toDto(),
+                        tekniskTid = Tidspunkt.now(),
+                        behandlingId = behandlingId,
+                    ),
+                )
+            }
+            VedtakStatus.IVERKSATT -> {
+                logger.warn(
+                    "Behandlet svar på samording for vedtak ${vedtak.id}, " +
+                        "men vedtaket er allerede iverksatt [behandling=$behandlingId. Skipper",
+                )
+            }
+            else -> {
+                verifiserGyldigVedtakStatus(vedtak.status, listOf(VedtakStatus.TIL_SAMORDNING))
+            }
+        }
+        return null
```

Comment:
Ser at det kalles to steder i route - hvor du har håndtert null.

Men brukes det ikke også i tilSamordningVedtak ?

```
            val vedtakEtterSvar = samordnetVedtak(behandlingId, brukerTokenInfo, tilSamordningVedtakLocal)
            return VedtakOgRapid(vedtakEtterSvar.vedtak, tilSamordning, vedtakEtterSvar.rapidInfo1)
```

Reply:
Jeg har forcet non-null der, da det er jo snarveien og det ikke skal være noe statusforvikling

Reply:
Spørs om man skal se på decoupling der også. Her og nå så ble jeg litt i stuss på hvorfor jeg ikke gjorde det fra start av. Kan være for å unngå litt ping pong mellom vedtaksvurdering  og vv-kafka

Reply:
OK - blir altid litt mistenksom om !! - men hvis vi er sikker :)